RepairsListener/UseCase/AddRepairsContractsUseCase.cs

```diff
@@ -0,0 +1,63 @@
+using RepairsListener.Boundary;
+using RepairsListener.Gateway.Interfaces;
+using RepairsListener.UseCase.Interfaces;
+using Hackney.Core.Logging;
+using System;
+using System.Threading.Tasks;
+using Hackney.Shared.Asset.Domain;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using RepairsListener.Infrastructure;
+
+namespace RepairsListener.UseCase
+{
+    public class AddRepairsContractsUseCase : IAddRepairsContractsUseCase
+    {
+        private readonly IRepairsStoredProcedureGateway _gateway;
+        private readonly ILogger<IAddRepairsContractsUseCase> _logger;
+
+
+        public AddRepairsContractsUseCase(IRepairsStoredProcedureGateway gateway, ILogger<IAddRepairsContractsUseCase> logger) : base()
+        {
+            _gateway = gateway;
+            _logger = logger;
+        }
+
+        [LogCall]
+        public async Task ProcessMessageAsync(EntityEventSns message)
+        {
+            if (message is null) throw new ArgumentNullException(nameof(message));
+
+            string jsonSnsMessage = JsonSerializer.Serialize(message);
+            _logger.LogInformation("RepairsListener received AssetCreatedEvent SNS message with body {JsonSnsMessage}", jsonSnsMessage);
+
+            // Get asset data from SNS message
+            string addRepairsContractsMessage = message.EventData.NewData.ToString();
+
+            // Required for deserialization
+            var options = new JsonSerializerOptions
+            {
+                PropertyNameCaseInsensitive = true
+            };
+
+            // Deserialize the JSON content directly into an AddRepairsContractsToNewAssetObject object
+            AddRepairsContractsToNewAssetObject addRepairsContractsMessageObject = JsonSerializer.Deserialize<AddRepairsContractsToNewAssetObject>(addRepairsContractsMessage, options);
+            _logger.LogInformation("AddRepairsContractsMessageObject is being processed. Object: {AddRepairsContractsMessageObject}", JsonSerializer.Serialize(addRepairsContractsMessageObject));
```

Comment:
I don't think it's necessary to serialise the object at each step, doing so impedes performance and is also a security risk as data is leaked to the logs each time - probably better to use log in ID

Reply:
There are two lines where I log out the serialized object and they're for troubleshooting purposes.
I'd prefer to leave these two logs there just for testing in dev, and remove them in another PR, once I make sure everything works as it should if that's ok

Reply:
OK